src/components/CommentsCard.jsx

```diff
@@ -1,7 +1,42 @@
+import { fetchCommentsByArticle } from "../../utils/api";
+import { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+
 export default function CommentsCard() {
+  const { article_id } = useParams();
+  const [comments, setComments] = useState();
+  const params = useParams();
+
+  useEffect(() => {
+    fetchCommentsByArticle(article_id).then((commentsData) => {
+      // console.log(typeof commentsData.comments);
+      setComments(commentsData.comments);
+    });
+  }, []);
+
+  console.log(params, "params");
+  console.log(comments, "comments");
+
+  if (!comments) {
+    return <div>Comment section loading, please wait.</div>;
+  }
+
```

Comment:
This would be better as an actual isLoading design pattern, so once the api call is successful, the comments state is filled with the comments from the api, and the isLoading state is changed. Currently it sort of works because the initial state of `comments` is undefined, which is falsy. But if you refer to my comment I made on line 7 about setting an initial state of [], then this section of code will stop working. I'd encourage you to implement a loading pattern and state to deal with this. 

Also, as a side comment, I think you'll be able to find a more appropriate semantic HTML tag which is not a `<div>`, please replace this div.